python/marvin/tools/query.py

```diff
@@ -1020,6 +1020,19 @@ def _build_query(self):
 
         # check if the query filter is against the DAP
         if self._check_for(self.filter_params.keys(), schema='dapdb'):
+
+            # Checks if the only table queried from dapdb is dapall. In that
+            # case we allow the query.
+            fparams = self._marvinform._param_form_lookup.mapToColumn(self.filter_params.keys())
+            tables = [c.class_.__table__.name for c in fparams
+                      if 'dapdb' in c.class_.__table__.schema]
+            all_dapall = all([tt == 'dapall' for tt in tables])
+
+            if not all_dapall and config.access not in config._dap_query_modes:
```

Comment:
why are you using `config.access` in places to check for dap query modes?  Was there a reason here or just a  hack since `dap_query_modes` is always an empty list and never looks like it gets populated?

Reply:
Everything is a hack. The idea is that we could decide whether a certain version of Marvin allows access to the DAP queries for a certain group of people (public or collaboration). `_dap_query_modes` will not change in runtime, only if we change it when tagging a new version. The idea is that in the future we may want to open DAP queries to the collaboration but not the public. Instead of changing that in several places, modifying `_dap_query_modes` in `__init__` would centralise it.

Really the current code won't allow you to do that. There is an inheriting problem with the use of that `query_params_metadata.cfg` that I hacked together. If we allowed the collaboration to use the DAP queries but not the public then the server would need to switch between `query_params_metadata.cfg` and `query_params_best.cfg`, which I haven't implemented.

The current version works (I think) in case that we don't allow anybody to access to DAP queries, which I think is what we decided we'd do if we decide to disable the DAP queries for now.

Reply:
Ok.  Just trying to understand.  I may need to change things a bit to fix the datamodels.  Those still get the full list of parameters so there are still some conflicts and bugs lingering about.   The `bestparams` are really only used in a couple places, to create the initial list of groups in the datamodels and to show a subset of what is available for the web.  The rest of the code uses the datamodel.  I'll try to work on it.